Sources/TuistCache/ContentHashing/TargetActionsContentHasher.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import TuistCore
+
+public protocol TargetActionsContentHashing {
+    func hash(targetActions: [TargetAction]) throws -> String
```

Comment:
I'd define a reference-type *(class)* and pass it down to cache the paths that have already been cached. Something along the lines of:

```swift
protocol CacheContentHashingCaching: AnyObject {
  func hash(path: AbsolutePath) throws -> String
  func hash<T: Hasheable>(item: T) throws -> String 
}
```
The class could be initialized with a `ContentHashing` instance:

```swift
final class CacheContentHashingCache: CacheContentHashingCaching {
  private let contentHasher: ContentHashing

  init(contentHasher: ContentHashing = ContentHasher()) {
    self.contentHasher = contentHasher
  }

  // ... Conformance of the CacheContentHashingCaching protocol
}
```

Then the method `hash` in all your `...Hasher` utilities can be something along the lines of:

```swift
func hash(targetActions: [TargetAction], cache: CacheContentHashingCaching) throws -> String
```

Note how we are using dependency injection through the public interface to pass an in-memory cache. This will give us the flexibility in the future to extend that cache beyond just memory and keep things stored in disk so that the cache is valid across runs.

Reply:
No need to do it as part of this PR, but I think is a good improvement to have before we keep architecting all the hashing business logic.

Reply:
Implemented!